src/executor_heredoc.c

```diff
@@ -0,0 +1,153 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   executor_heredoc.c                                 :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/08/05 11:05:42 by sqiu              #+#    #+#             */
+/*   Updated: 2023/08/06 23:25:34 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file executor_heredoc.c
+ * @brief Functions to handle heredoc feature.
+ */
+
+#include "mod_executor.h"
+
+/**
+ * @brief Drive creation of multiple heredocs.
+ * 
+ * @param cmd 		Current cmd being processed.
+ * @return t_err 	ERR_MALLOC, ERR_CLOSE, ERR_BAD_FD, SUCCESS
+ */
+t_err	ft_handle_heredoc(t_cmd *cmd)
+{
+	int		i;
+	t_err	err;
+
+	while (cmd)
+	{
+		i = -1;
+		while (++i < cmd->delim_pos)
+		{
+			err = ft_create_heredoc(cmd, cmd->delims[i], i);
```

Comment:
The heredoc name is created with cmd->index. See line 64:

err = ft_initiate_heredoc(cmd->index, &name, &fd);

The counter variable i serves to know whether the last heredoc has been reached (in case of several heredocs in a cmd). In that case the heredoc is not unlinked, but retained and its fd assigned to cmd->fd_in